File: CSharp-Part-1/4. Console In and Out/ConsoleInAndOut/QuadraticEquation/QuadraticEquation.cs
// Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).

using System;

    class QuadraticEquation
    {
        static void Main()
        {
            Console.WriteLine("Enter a: ");
            double a = double.Parse(Console.ReadLine());
            Console.WriteLine("Enter b: ");
            double b = double.Parse(Console.ReadLine());
            Console.WriteLine("Enter c: ");
            double c = double.Parse(Console.ReadLine());

            if (a == 0)
            {
                // The equation is linear
                Console.WriteLine("x1 = x2 = {0}", c/b);
            }
            else
            {
                // The equation is indeed quadratic
                double discriminant = Math.Pow(b, 2) - 4*a*c;
                if (discriminant < 0)
                {
                    Console.WriteLine("No real roots.");
                }
                else if (discriminant == 0)
                {
                    Console.WriteLine("x1 = x2 = {0}", -b/2*a);
                } else {
                    Console.WriteLine("x1 = {0}", (-b + Math.Sqrt(Math.Pow(b, 2) - 4*a*c)) / 2*a);
                    Console.WriteLine("x2 = {0}", (-b - Math.Sqrt(Math.Pow(b, 2) - 4 * a * c)) / 2 * a);
                }
            }
        }
    }

File: JavaScript-Fundamentals/05. Conditional Statements/quadratic-equation.js
/*
	Write a script that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).
	Calculates and prints its real roots.
	
	Note: Quadratic equations may have 0, 1 or 2 real roots.
*/

function solveQuadraticEquation(a, b, c) {
	if (a === 0) {
		return -c / b;
	} else {
		discriminant = b * b - 4 * a * c;
		if (discriminant < 0 ) {
			return 'no real roots';
		} else {
			if (discriminant === 0) {
				x = -b / (2 * a);
				return 'x1 = x2 = ' + x;
			} else {
				x1 = (-b - Math.sqrt(discriminant)) / (2 * a);
				x2 = (-b + Math.sqrt(discriminant)) / (2 * a);
				return 'x1 = ' + x1 + ' ; ' + 'x2 = ' + x2;
			}
		}
	}
}

console.log(solveQuadraticEquation(2, 5, -3));
console.log(solveQuadraticEquation(-1, 3, 0));
console.log(solveQuadraticEquation(-0.5, 4, -8));
console.log(solveQuadraticEquation(5, 2, 8));